common/changes/@microsoft/rush/feat-pnpm-patch_2022-07-27-03-39.json

```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "rush-pnpm accepts patch and patch-commit",
+      "type": "none"
+    }
+  ],
```

Comment:
@elliot-nelson wrote:
> The only test that did _not_ work was this one:
> 
>     5. _Can you target any package?_ - **No**. For example, from the root of the monorepo, you can `rush-pnpm patch` any transitive dependency at any level of the lockfile, but when you try to _apply_ that patch with `rush-pnpm patch-commit`, it will fail. Similarly, you can `rush-pnpm patch` any package from a project folder (even one that isn't in that package.json), but trying to apply the patch will fail.
> 
> 
> (I don't believe this is a blocker, and maybe it's something that can be fixed in future. The result here is that if you want to "target" a package deep in your lockfile, you first need to _add_ it to a package.json file somewhere in your monorepo to make it visible, then you can patch and apply the patch, and then remove it from that project.)

I encountered this same problem.  Since I'm patching `is-odd` for the entire monorepo, why do I need to invoke it under a project folder?

The error message itself isn't super intuitive:

```
 ERR_PNPM_NO_PKG_MANIFEST  No package.json found in C:\rush-pnpm-patch-demo
 ```

How would we improve this though?  Is it really a design problem of PNPM?

Reply:
I believe this PR https://github.com/microsoft/rushstack/pull/3791 can fix this issue 🤞